opencog/guile/SchemeSmobTV.cc

```diff
@@ -460,5 +460,32 @@ SCM SchemeSmob::ss_tv_get_value (SCM s)
 	return SCM_EOL;
 }
 
+/**
+ * Return the truth value mean
+ */
+SCM SchemeSmob::ss_tv_get_mean(SCM s)
+{
+	TruthValue *tv = verify_tv(s, "cog-tv-mean");
+	return scm_from_double(tv->getMean());
+}
```

Comment:
OK, I understand. So what I should do is have one C++ binding for {mean, confidence, count}, and then use scheme helpers to extract them.

Alternatively, I can add these mean, confidence, count fields to the indefinite TV scheme alist representation, which would make the tv-mean, tv-conf and tv-count work. There's a catch though, requesting the count of an itv might be computationally expensive (it isn't today, but the algorithm is wrong), and so is better done only when necessary.

So I'm a bit puzzled about what to do.

But wait!!! cog-tv->alist isn't memoized, is it? If not then calling each once tv-mean, tv-conf and tv-count should be roughly equivalent to calling cog-tv-mean, cog-tv-conf and cog-tv-count (even slightly faster), since tv-mean, etc rely on cog-tv->alist.